Mathlib/Tactic/NormNum/BigOperators.lean

```diff
@@ -0,0 +1,372 @@
+/-
+Copyright (c) 2023 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Floris van Doorn
+-/
+import Mathlib.Tactic.NormNum.Basic
+import Mathlib.Algebra.BigOperators.Basic
+import Mathlib.Data.List.FinRange
+
+/-!
+# `norm_num` plugin for big operators
+
+This file adds `norm_num` plugins for `Finset.prod` and `Finset.sum`.
+
+The driving part of this plugin is `Mathlib.Meta.NormNum.evalFinsetBigop`.
+We repeatedly use `Finset.proveEmptyOrCons` to try to find a proof that the given set is empty,
+or that it consists of one element inserted into a strict subset, and evaluate the big operator
+on that subset until the set is completely exhausted.
+
+## Porting notes
+
+This plugin is noticeably less powerful than the equivalent version in Mathlib 3: the design of
+`norm_num` means plugins have to return numerals, rather than a generic expression.
+In particular, we can't use the plugin on sums containing variables.
+(See also the TODO note "To support variables".)
+
+## TODOs
+
+ * Support intervals: `Finset.Ico`, `Finset.Icc`, ...
+ * To support variables, like in Mathlib 3, turn this into a standalone tactic that unfolds
+   the sum/prod, without computing its numeric value (using the `ring` tactic to do some
+   normalization?)
+-/
```

Comment:
Could we have a cross-reference to the `fin_cases` tactic, which has related scope?